### PR TITLE
Even more `db.DefaultContext` refactor

### PR DESCRIPTION
**Pull request by JakobDev**
Part of #27065

### CODE REVIEW

File: models/issues/issue_watch.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func CheckIssueWatch(ctx context.Context, user *user_model.User, issue *Issue) (
 	if err != nil {
 		return false, err
 	}
-	return repo_model.IsWatchMode(w.Mode) || IsUserParticipantsOfIssue(user, issue), nil
+	return repo_model.IsWatchMode(w.Mode) || IsUserParticipantsOfIssue(db.DefaultContext, user, issue), nil
```
Comment (delvh) [inline models/issues/issue_watch.go]: I think you mean
```suggestion
	return repo_model.IsWatchMode(w.Mode) || IsUserParticipantsOfIssue(ctx, user, issue), nil
```
?

File: routers/web/repo/issue.go [hunk 2]
```diff
@@ -1549,8 +1549,8 @@ func ViewIssue(ctx *context.Context) {
 		} else {
 			ctx.Data["CanUseTimetracker"] = false
 		}
-		if ctx.Data["WorkingUsers"], err = issues_model.TotalTimesForEachUser(&issues_model.FindTrackedTimesOptions{IssueID: issue.ID}); err != nil {
-			ctx.ServerError("TotalTimesForEachUser", err)
+		if ctx.Data["WorkingUsers"], err = issues_model.TotalTimesForEachUser(ctx, &issues_model.FindTrackedTimesOptions{IssueID: issue.ID}); err != nil {
+			ctx.ServerError("TotalTimes", err)
```
Comment (delvh) [inline routers/web/repo/issue.go]: I think something got mangled through the recent PR here.

Comment (delvh) [inline routers/web/repo/issue.go]:
```suggestion
			ctx.ServerError("TotalTimesForEachUser", err)
```

**Comment by delvh** [comment]
Should I apply the suggestions myself?

File: models/issues/issue.go [hunk 3]
```diff
@@ -201,12 +201,12 @@ func (issue *Issue) IsTimetrackerEnabled(ctx context.Context) bool {
 }
 
 // GetPullRequest returns the issue pull request
-func (issue *Issue) GetPullRequest() (pr *PullRequest, err error) {
+func (issue *Issue) GetPullRequest(ctx context.Context) (pr *PullRequest, err error) {
```
Comment (lunny) [inline models/issues/issue.go:204]: `templates/shared/issueicon.tmpl` need to be updated

**Review by delvh — APPROVED** [review]
LGTM as soon as @lunny's request has been implemented.

File: models/issues/issue.go (hunk 3 above)
Comment (JakobDev) [reply models/issues/issue.go:204]: God catch! We really need better frontend tests.

**Review by lunny — APPROVED** [review]

**Pull request merged**